### PR TITLE
feat(BITESIZE-8826): Add sitemap location override

### PR DESCRIPTION
**Pull request by danieleloscozzese**
This allows the sitemap fragments to be specifically located using an absolute URL, for the case where a sitemap index needs to be generated.

We do not want our sitemaps to be hosted at the root of our site, which is the only location that the current code writes the `<loc>` entries to point to when there is more than 1 part of the sitemap and an index is generated.

There were no behavioural tests so I've managed to smash something together that allows the result of the `'complete'` event to be tested, so that the behaviour of the override can be verified.

### CODE REVIEW

**Review by JackRichardM — APPROVED** [review]
Looks good to me 👍

**Review by brittvarnom — APPROVED** [review]
LGTM!

File: src/__tests__/index.js [hunk 1]
```diff
@@ -1,4 +1,57 @@
-const SitemapGenerator = require('../');
+const cheerio = require('cheerio');
```
Comment (Karthik-Y) [inline src/__tests__/index.js:1]: Is cheerio already added? I don't see it in the package json. Between nice work on the tests.

**Review by Karthik-Y — APPROVED** [review]
Looks good to me, but a healthy eyes is recommended.

File: src/__tests__/index.js (hunk 1 above)
Comment (danieleloscozzese) [reply src/__tests__/index.js:1]: It was there as a main dependency already, yeah.

**Pull request merged**